docs/vsock.md

```diff
@@ -0,0 +1,147 @@
+# VSOCK support
+
+VSOCKs provide a way for guest and host to communicate through a socket. VSOCK sockets support both stream and datagram types.
+
+# Basic Terms
+
+## C Definitions
+
+The below C definitions illustrate the data structures behind VSOCK and are helpful to understand the calculations used in the following examples.
+
+```C
+// bits/socket.h
+#define PF_VSOCK        40      /* vSockets.  */
+```
+
+```C
+// linux/socket.h
+typedef unsigned short __kernel_sa_family_t;
+```
+
+```C
+// linux/vm_sockets.h
+struct sockaddr_vm {
+        __kernel_sa_family_t svm_family;
+        unsigned short svm_reserved1;
+        unsigned int svm_port;
+        unsigned int svm_cid;
+        __u8 svm_flags;
+        unsigned char svm_zero[sizeof(struct sockaddr) -
+                               sizeof(sa_family_t) -
+                               sizeof(unsigned short) -
+                               sizeof(unsigned int) -
+                               sizeof(unsigned int) -
+                               sizeof(__u8)];
+};
+```
+
+The `sizeof(struct sockaddr)` is 16 bytes, thus the size of the `svm_zero` element would equal 3 in this case.
+Note, that the given struct layout needs to be verified on a concrete system. The stucts shown here have been observed
+on a system with glibc 2.35 and kernel 5.17. Be aware, that depending the libc version, the struct layout can differ.
```

Comment:
Not sure how relevant this whole section is.

Reply:
Yep, probably it goes to deep into the glibc level. I've just added the below to the socat notes instead, just in case.

Consult `linux/vm_sockets.h` on the given system for the details of `struct sockaddr_vm` layout.

Thanks